modules/effects/src/concat_latest_from.ts

```diff
@@ -29,9 +29,9 @@ export function concatLatestFrom<
       const observablesAsArray = Array.isArray(observables)
         ? observables
         : [observables];
-      return of(value).pipe(
+      return (of(value).pipe(
```

Comment:
By taking a look at the new RxJS types for `withLatestFrom`, I think we can simplify this.
Maybe in another PR though (because it might be breaking if end-users were using the generics).

```ts
export function concatLatestFrom<T, O extends unknown[], V>(
  observablesFactory: (value: V) => [...ObservableInputTuple<O>]
): OperatorFunction<T, [T, ...O]>;
export function concatLatestFrom<T, O, V>(
  observablesFactory: (value: V) => ObservableInput<O>
): OperatorFunction<T, [T, O]>;

/**
 * `concatLatestFrom` combines the source value
 * and the last available value from a lazily evaluated Observable
 * in a new array
 */
export function concatLatestFrom<T, R>(
  observablesFactory: (value: any) => any | any[]
): OperatorFunction<T, R | any[]> {
  return pipe(
    concatMap((value) => {
      const observables = observablesFactory(value);
      const observablesAsArray = Array.isArray(observables)
        ? observables
        : [observables];
      return of(value).pipe(withLatestFrom(...observablesAsArray));
    })
  );
}
```

Reply:
that's pretty cool!